### PR TITLE
Update which go modules have a version() func

### PR DESCRIPTION
**Pull request by zhyChesterCheung**
Resolved: #2143 
- Maintain a version.go file to define the Version() func for every module within the repo
- Add test func for each func

### CODE REVIEW

File: sdk/metric/version_test.go
Comment (MrAlias) [inline sdk/metric/version_test.go:25]: Can we move this `var` to https://github.com/open-telemetry/opentelemetry-go/tree/main/internal/internaltest so it isn't repeated everywhere?

Maybe also add a version testing function there as well?

Comment (zhyChesterCheung) [reply sdk/metric/version_test.go]: Already done as above

File: CHANGELOG.md
Comment (MrAlias) [inline CHANGELOG.md:14]: Can we remove this? These are not exported changes and we probably do not need to communicate them to the end user.

File: internal/internaltest/version.go
Comment (MrAlias) [inline internal/internaltest/version.go:29]: I might not have been clear in my last suggestion that we should add a version testing function. I was anticipating that we would unify the `TestVersionSemver` functions here. Adding the helper function isn't a must do, but I would remove this before merging these changes.

File: sdk/trace/version.go
Comment (MrAlias) [inline sdk/trace/version.go:19]:
```suggestion
	return "1.0.0-RC2"
```

File: sdk/metric/version.go
Comment (MrAlias) [inline sdk/metric/version.go:17]:
```suggestion
// version is the current release version of the metric SDK in use.
```

File: sdk/trace/version.go
Comment (MrAlias) [inline sdk/trace/version.go:17]:
```suggestion
// version is the current release version of the trace SDK in use.
```

File: internal/internaltest/version.go
Comment (zhyChesterCheung) [reply internal/internaltest/version.go]: I'v done that as followed. Please help to review it again.

File: go.mod
Comment (eddyleelin) [inline go.mod:9]: Sorry, do you know why this sdk/metric module needs to be added to the require dependencies of go.mod? I don't think that would be right, since then the stable top-level "go.opentelemetry.io/otel" (in module set stable-v1) would have a dependency on the unstable module `go.opentelemetry.io/otel/sdk/metric`. Or is it just a placeholder for something?

File: sdk/metric/version.go
Comment (eddyleelin) [inline sdk/metric/version.go:19]: This shouldn't have the "v" in front of the version number, right? The current root-level `version.go` and the trace file don't have the "v"?

File: internal/internaltest/version_test.go
Comment (eddyleelin) [inline internal/internaltest/version_test.go:42]: This might need to be changed to be consistent with my other comment?

File: internal/internaltest/version.go
Comment (eddyleelin) [inline internal/internaltest/version.go:22]: This test is looking for a "v" in front of the version number, is that the intended behavior?

**Review by eddyleelin — CHANGES_REQUESTED** [review]
Nice one, thanks for the help with this issue! I just wanted to make sure that we're on the same page with a couple of things?

- Should version numbers start with a "v" (asked in the comments below)?
- Also, I'm not 100% sure, but did we decide in the SIG meeting to remove the version.go file in the repo root, or will we be keeping that there?

File: internal/internaltest/version_test.go
Comment (Aneurysm9) [inline internal/internaltest/version_test.go:21]: The API cannot depend on the SDK.  If this testing needs to happen then it should happen in the SDK packages using helpers from this package.

**Review by Aneurysm9 — CHANGES_REQUESTED** [review]

File: go.mod
Comment (jmacd) [reply go.mod:9]: (Agree, this looks incorrect.)

**Review by MadVikingGod — CHANGES_REQUESTED** [review]
This currently tests that the versions returned by the modules are semvers, but I don't see how useful that is.  It would be much more useful if we tested that the versions match what is in our go.mod or something along those lines.

I would approve if we fix where this lives (the import problem) and if some of the non-semver strings where excluded.

**Comment by MrAlias** [comment]
This is quite stale at this point, and it seems to have been abandoned. I'm going to close it for these reasons. Please feel free to address the issues pointed out here and resubmit.

**Pull request closed without merging**
